### PR TITLE
Variadic arguments for tf.control_dependencies

### PR DESCRIPTION
**Pull request by carlthome**
# Disclaimer: this is a UX proposal for the Python API but it should be discussed first!

## Proposal
I see a lot of people get tripped up on `tf.control_dependencies` as they often try to pass a single tensor/op to it and forget to explicitly wrap it in a Python list. Perhaps it would be better to support both usages, as that is more in line with how `tf.keras` does things.

**TL;DR: This commit changes `tf.control_dependencies` so that it takes a list as before, but also works with variadic arguments as well.**

```py
with tf.control_dependencies(tf.assert_positive(x)):
  x = tf.identity(x)
```

### CODE REVIEW

**Comment by josh11b** [comment]
API owner comment: Should use python.util.nest.flatten() to accept arbitrary nesting of lists, tuples, etc. Don't worry about collections.Iterable, flatten() will handle the common sequence types. Otherwise this looks good.

**Comment by carlthome** [comment]
Cool. `nest.flatten` looks very clean. 🎉

File: tensorflow/python/framework/ops.py [hunk 1]
```diff
@@ -4399,7 +4400,7 @@ def _record_op_seen_by_control_dependencies(self, op):
     for controller in self._control_dependencies_stack:
       controller.add_op(op)
 
-  def control_dependencies(self, control_inputs):
+  def control_dependencies(self, *control_inputs):
```
Comment (asimshankar) [inline tensorflow/python/framework/ops.py]: The docstring for control_inputs could use an update :)

Also, doesn't this break the behavior of `tf.control_dependencies(None)`?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by martinwicke** [comment]
@carlthome can you update the PR to address @asimshankar's comments?

**Comment by martinwicke** [comment]
@asimshankar another round of review?

**Review by asimshankar — CHANGES_REQUESTED** [review]

File: tensorflow/python/framework/ops.py [hunk 2]
```diff
@@ -4617,8 +4629,12 @@ def my_func(pred, tensor):
       TypeError: If `control_inputs` is not a list of `Operation` or
         `Tensor` objects.
     """
-    if control_inputs is None:
+    control_inputs = nest.flatten(control_inputs)
+    if all(x is None for x in control_inputs):
       return self._ControlDependenciesController(self, None)
+    elif any(x is None for x in control_inputs):
+      raise ValueError("To clear existing dependencies, every element passed "
```
[the remark this answers is not in the document]
Comment (carlthome) [reply tensorflow/python/framework/ops.py]: Is there a better term for "regular object" than "non-None" that's Pythonic?

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**